pkg/certwatcher/certwatcher_test.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certwatcher_test
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"net"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/certwatcher"
+)
+
+var _ = Describe("CertWatcher", func() {
+	var _ = Describe("certwatcher New", func() {
+		It("should errors without cert/key", func() {
+			_, err := certwatcher.New("", "")
+			Expect(err).ToNot(BeNil())
+		})
+	})
+
+	var _ = Describe("certwatcher Start", func() {
+		var (
+			ctx       context.Context
+			ctxCancel context.CancelFunc
+			watcher   *certwatcher.CertWatcher
+		)
+
+		BeforeEach(func() {
+			ctx, ctxCancel = context.WithCancel(context.Background())
+
+			err := writeCerts(certPath, keyPath, "127.0.0.1")
+			Expect(err).To(BeNil())
+
+			Eventually(func() error {
+				for _, file := range []string{certPath, keyPath} {
+					_, err := os.ReadFile(file)
+					if err != nil {
+						return err
+					}
+					continue
+				}
+
+				return nil
+			}).Should(Succeed())
+
+			watcher, _ = certwatcher.New(certPath, keyPath)
```

Comment:
Why the unhandled error?

Reply:
no particular reason, I can add and `Expect` here as well.

Reply:
Updated.